File: 1 - Foundations/1.3 - Designing Algorithms/Code/bubble_sort.py
"""
bubble_sort.py

"""


def bubble_sort(array):
    """Bubble Sort Algorithm. Popular, but inefficent as it runs in quadratic time

    Time complexity: O(n^2)
    Space complexity: O(1)

    Args:
        array (list[int]): Input array of integers
    """
    n = len(array)

    for i in range(0, n):
        for j in range(n - 1, i, -1):
            if array[j] < array[j - 1]:
                array[j], array[j - 1] = array[j - 1], array[j]

File: 1 - Foundations/1.3 - Designing Algorithms/Code/BubbleSort.mjs
/*
 * BubbleSort.mjs
 *
 */

function bubbleSort(array) {
    let n = array.length;

    for (let i = 0; i < n; i++) {
        for (let j = n - 1; j > -1; j--) {
            if (array[j] < array[j - 1]) {
                let temp = array[j];
                array[j] = array[j - 1];
                array[j - 1] = temp;
            }
        }
    }
}

const main = () => {
    var array = [2, 4, 6, 7, 1, 2, 3, 5, 8, 10, 9];
    bubbleSort(array);

    console.log(array);
};

// main();

export default bubbleSort();
